nucypher/network/server.py

```diff
@@ -230,6 +233,28 @@ def set_policy(id_as_hex):
             # TODO: Perhaps we log this?
             return Response(status_code=400)
 
+        if not this_node.federated_only:
+            # This splitter probably belongs somewhere canonical.
+            transaction_splitter = BytestringSplitter(32)
+            tx, cleartext = transaction_splitter(cleartext, return_remainder=True)
+
+            try:
+                # Get all of the arrangements and verify that we'll be paid.
+                # TODO: We'd love for this part to be impossible to reduce the risk of collusion.  #1274
+                arrangements = this_node.policy_agent.fetch_arrangements_from_policy_txid(tx, timeout=20)
+            except TimeExhausted:
+                # Alice didn't pay.  Return response with that weird status code.
+                this_node.suspicious_activities_witnessed['freeriders'].append((alice, f"No transaction matching {tx}."))
```

Comment:
Easily one of the most `suspicious_activities` in the code base hahaha

Reply:
This one isn't so terrible, it's the one below that can only possible be absolutely malicious.